src/routers/datatypeRouters/edges/go_terms_annotations.ts

```diff
@@ -0,0 +1,150 @@
+import { z } from 'zod'
+import { publicProcedure } from '../../../trpc'
+import { loadSchemaConfig } from '../../genericRouters/genericRouters'
+import { RouterEdges } from '../../genericRouters/routerEdges'
+import { paramsFormatType } from '../_helpers'
+import { RouterFilterBy } from '../../genericRouters/routerFilterBy'
+import { descriptions } from '../descriptions'
+import { db } from '../../../database'
+import { QUERY_LIMIT } from '../../../constants'
+
+const schema = loadSchemaConfig()
+
+const transcriptSchema = schema.transcript
+const proteinSchema = schema.protein
+
+const goTermsAnnotationsSchema = schema.gaf
+const goAnnotationsRouter = new RouterEdges(goTermsAnnotationsSchema)
+
+const goTermQueryFormat = z.object({
+  go_term_id: z.string(),
+  page: z.number().default(0)
+})
+
+const queryFormat = z.object({
+  query: z.string(),
+  page: z.number().default(0)
+})
+
+const goAnnotationFormat = z.object({
+  annotation_id: z.string(),
+  annotation_name: z.string(),
+  go_term_name: z.string(),
+  source: z.string(),
+  gene_product_type: z.string(),
+  gene_product_symbol: z.string(),
+  qualifier: z.array(z.string()),
+  organism: z.string(),
+  evidence: z.string(),
+  go_id: z.string()
+}).optional()
+
+async function transcriptIds (id: string): Promise<any[]> {
+  const input: paramsFormatType = {}
+  input.transcript_name = id
+  input.transcript_id = id
+  input._key = id
+
+  return await (new RouterFilterBy(transcriptSchema)).getObjectIDs(input, '', false)
+}
+
+async function proteinIds (id: string): Promise<any[]> {
+  const input: paramsFormatType = {}
+  input.name = id
+  input.dbxrefs = id
+  input._key = id
+
+  return await (new RouterFilterBy(proteinSchema)).getObjectIDs(input, '', false)
+}
+
+const goTermAnnotationsCollection = goTermsAnnotationsSchema.db_collection_name
+
+async function goTermsSearch (input: paramsFormatType): Promise<any[]> {
+  const query = input.query as string
+  const page = input.page as number
+
+  let annotations: string[]
+
+  // assuming an ID will match either transcripts or proteins
+  const transcripts = await transcriptIds(query)
+  if (transcripts.length !== 0) {
+    annotations = transcripts
+  } else {
+    annotations = await proteinIds(query)
+  }
+
+  if (annotations.length > 0) {
+    const query = `
```

Comment:
we've decided to start a big refactoring and start bringing the queries from the routerEdge file into each independent router so it's easier to read and maintain. We will keep the routerEdges file as is for now in the project in case it's needed in the future